### DISCUSSION TITLE
Release v1.0

### DISCUSSION THREAD

**Discussion by jarabroz** [Dlubal-Software/RFEM_Python_Client#53] [General]
The first release of Python High Level Function Library

<hr /><em>This discussion was created from the release <a href='https://github.com/Dlubal-Software/RFEM_Python_Client/releases/tag/'>Release v1.0</a>.</em>

**Comment by ealgit** [comment]
Hi. Will you release the C# library any time soon? :-)

**Reply by jarabroz** [reply]
Hi,
thank you @ealgit for your question. We are currently working on out C#. We have a system hoe to generate sucha library from WSDL definition. We want to produce this library based on  .Net Core 5.0. We are in the middle of the migration. It will be released soon.

**Reply by michaelvantelgen-work** [reply]
Hi @jarabroz,

I too am looking forward to seeing the C# library. 

Can I ask a couple of questions about it? Because I will need to migrate a couple of inhouse tools from RFEM5 to RFEM6 when it is available:

1. Will the C# library work like the RFEM5 library in the sense that you import it as a reference in your application? 
2. Will the functions work like we are used to? (for instance that you define an object, set its properties and write it to RFEM)
3. Will that be an open source development too? If so, I would love to be part of a community of developers that can learn from each other.
4. Could you let us know here or elsewhere when it is released?

Looking forward to your response and the library! 

Kind regards, 
Michael van Telgen

**Reply by jarabroz** [reply]
Hi @michaelvantelgen-Arcadis ,

Thank you for your questions. Sure, you can ask. I'm glad to hear you are looking at our solution and thinking about migration of tools from RFEM5 to RFEM6. I'm here to help you with it.

1.  Yes, it will be like solution with RFEM5. You just need to reference dll and fulfill requirements for dependencies. We would like to publish Nuget package for better distribution and maintenance of the dlls.
2. Yes, our library contains classes for objects and methods for get and set to RFEM6. Each class has publicly accessible properties which you can use.
3. Yes, we will publish our C# library as open source under MIT licence as we did for Python. So that, community can contribute.
4. We are going to use the same as for Python - GitHub repository. We are finalizing the C# library so that it could be use cross-platform. We want to have .Net Core 5.0. I expect that we will release it in the middle of November.

Kind  regards

Jaroslav Broz

**Reply by michaelvantelgen-work** [reply]
Hi @jarabroz ,

Thanks for your swift reply! NuGet would be great and it sound like it will not be a difficult migration then. 

I'm already looking forward to it!

Kind regards,
Michael van Telgen

**Reply by ealgit** [reply]
Hi @jarabroz 
In our company we are using both RFEM and RSTAB. Will there be seperate releases for RSTAB9?

Kind regards,
Edin

**Reply by jarabroz** [reply]
Hi @ealgit ,
in fact, you can use Python library for RFEM and RSTAB. For C# we will have separate library for RSTAB9.
